File: problem solving/Implement Queue using Stacks/Implement Queue using Stacks.cpp
class MyQueue {
private:
    deque<int> d;
public:
    MyQueue() {
        
    }
    
    void push(int x) {
        d.push_back(x);
    }
    
    int pop() {
        int x=d.front();
        d.pop_front();
        return x;
    }
    
    int peek() {
        return d.front();
    }
    
    bool empty() {
        return d.size() < 1;
    }
};

File: problem solving/Implement Queue using Stacks/Implement Queue using Stacks.js
let deq=[];

var MyQueue = function() {
    deq=[];
};

MyQueue.prototype.push = function(x) {
    deq.push(x);
};

MyQueue.prototype.pop = function() {
    return deq.shift();
};

MyQueue.prototype.peek = function() {
    return deq[0];
};

MyQueue.prototype.empty = function() {
    return !deq.length;
};